Makefile

```diff
@@ -31,7 +33,18 @@ test:
 
 .PHONY: build
 build:
-	GOOS=$(GOOS) GOARCH=$(GOARCH) go build -o signoz-collector ./cmd/signozcollector
+	CGO_ENABLED=1 go build -tags timetzdata -o .build/${GOOS}-${GOARCH}/signoz-collector -ldflags "-linkmode external -extldflags '-static' -s -w ${LD_FLAGS}" ./cmd/signozcollector
```

Comment:
Was there a reason why `CGO_ENABLED` changed?

Reply:
Till Go `1.19`, `CGO_ENABLED` is enabled by default.
Since we were using alpine base image for both build time and run time,
we did not face any issues even though we had `ENV CGO_ENABLED=0` hardcoded in Dockerfile.

In case we set to `CGO_ENABLED` is false and build the static binary.
We get the following `core dumped` error when trying to run it.

```
[1]    2090581 trace trap (core dumped)  .build/linux-amd64/signoz-collector
```

The default behaviour of cgo is altered from Go 1.20. Hence, it is safer to enable
it on our own rather than relying on the defaults. Ref: https://tip.golang.org/doc/go1.20#cgo